JASP-Engine/JASP/R/multinomialtestbayesian.R

```diff
@@ -0,0 +1,446 @@
+#
+# Copyright (C) 2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+MultinomialTestBayesian <- function(jaspResults, dataset, options, state = NULL){
+
+  # Set title
+  jaspResults$title <- "Bayesian Multinomial Test"
+
+  # Read dataset
+  dataset <- .multinomialBayesReadData(dataset, options)
+
+  # Error checking
+  .multinomialBayesCheckErrors(dataset, options)
+
+  # Compute Results for Tables
+  multinomialResults <- .computeMultinomialResults(jaspResults, dataset, options)
+  # Output tables and plots
+  .createMultBayesMainTable(jaspResults, options, multinomialResults)
+  .createMultBayesDescriptivesTable(jaspResults, options, multinomialResults)
+  .createMultBayesDescriptivesPlot(jaspResults, options, multinomialResults)
+
+  return()
+
+}
+
+# Results functions
+.computeMultinomialResults <- function(jaspResults, dataset, options) {
+
+  # Take results from state if possible
+  if (!is.null(jaspResults[["stateMultBayesResults"]])) return(jaspResults[["stateMultBayesResults"]]$object)
+
+  # This will be the object that we fill with results
+  multinomialResults <- list(mainTable         = list(),
```

Comment:
I really think this function should just be split up in separate functions for each output element, but that's stylistic.